flink-end-to-end-tests/test-scripts/test_ha_dataset.sh

```diff
@@ -53,20 +52,51 @@ function run_ha_test() {
 
     wait_job_running ${JOB_ID}
 
-    # start the watchdog that keeps the number of JMs stable
-    start_ha_jm_watchdog 1 "StandaloneSessionClusterEntrypoint" start_jm_cmd "8081"
-
+    local c
     for (( c=0; c<${JM_KILLS}; c++ )); do
         # kill the JM and wait for watchdog to
         # create a new one which will take over
         kill_single 'StandaloneSessionClusterEntrypoint'
         wait_job_running ${JOB_ID}
     done
 
-    cancel_job ${JOB_ID}
+    for (( c=0; c<${TM_KILLS}; c++ )); do
+        sleep $(( ( RANDOM % 10 )  + 1 ))
+        kill_and_replace_random_task_manager
+        wait_job_running ${JOB_ID}
+    done
+
+    wait_job_terminal_state ${JOB_ID} "FINISHED"
```

Comment:
I'm wondering how useful this test can be if it has a finite runtime. For all we know the job could finish before we even kill the JM, or maybe more likely, any TM.

From what we've seen so far, this can only result in either:
a) The test taking significantly more time than required, as we have to ensure that all these events happen before it terminates
b) The test being unstable.

I like neither option, do admit though that this would make it very difficult (or even impossible) to verify the correctness of the output.

How much longer does the test now run for?

It looks like the test was _supposed_ to run indefinitely (`--source true`), but this isn't in sync with the actual parameters of the job which use `infinite` as the parameter.

Reply:
These are valid concerns.

> How much longer does the test now run for?

The test runs 4.5-5 minutes on my machine. It takes around 2 minutes to complete the batch job after the last injected fault (time determined using unscientific methods). 

> I like neither option, do admit though that this would make it very difficult (or even impossible) to verify the correctness of the output.

I don't see a good solution yet. Here are some options:
1. Make job block on external signals (files)
1. Leave it as before, i.e., don't verify correctness of the output

Reply:
I added a new job that blocks on an external condition.